tests/integration/json_test.go

```diff
@@ -83,9 +83,11 @@ var _ = Describe("odojsonoutput", func() {
 			// odo list -o json --path .
 			pwd := helper.Getwd()
 			desired := fmt.Sprintf(`{"kind":"List","apiVersion":"odo.openshift.io/v1alpha1","metadata":{},"items":[{"kind":"Component","apiVersion":"odo.openshift.io/v1alpha1","metadata":{"name":"nodejs","creationTimestamp":null},"spec":{"type":"nodejs","source":"https://github.com/openshift/nodejs-ex"},"status":{"context":"%s","state":"Pushed"}}]}`, strings.TrimSpace(pwd))
-			helper.Chdir("/tmp")
+			tmpDir := helper.CreateNewContext()
+			helper.Chdir(tmpDir)
 			actual := helper.CmdShouldPass("odo", "list", "-o", "json", "--path", pwd)
 			helper.Chdir(pwd)
+			helper.DeleteDir(tmpDir)
```

Comment:
if the `CmdShouldPass` fails the `tmpDir` is not going to be deleted :-(

Reply:
call `defer helper.DeleteDir(tmpDir)` right after `tmpDir := helper.CreateNewContext()`

Reply:
updated